### ISSUE TITLE
`forwardDateOnly` is a confusingly titled option

### ISSUE THREAD

**Issue by nbikar** [wanasit/chrono#132]
text="Mar 2-6"
var results = chrono.parse(text,new Date(),{forwardDateOnly: true});

returns 

start.date() =Mar 2, 2016
end.date() = Mar 6, 2016

However if text="Feb 1-4"

start.date() = Feb 1, 2017
end.date() = Feb 4, 2017

What am I missing?

On a similar note, 

if text="Sat -Wed"    (today is Friday)

start.date() = Wed, Aug 3,2016    (date in the past)
end.date() = Sat, Aug 6, 2016

**Comment by penne12** [comment]
`forwardDateOnly` only works for scenarios like the one in issue #118. The naming is a bit unclear, sorry. You could make a custom refiner to implement this, though, or send a PR.

**Comment by wanasit** [comment]
@penne12 is right. The `forwardDateOnly` is used only for day-of-week guessing. but maybe think I could fix this though.

**Comment by penne12** [comment]
@wanasit For the time being, should we rename the option `forwardDateOnly` and deprecate the current naming?

**Comment by wanasit** [comment]
I have some time today. Let me try fixing that. 
I think we could have a refiner that try re-guessing the time or year on `forwardDateOnly` option.

**Comment by penne12** [comment]
Sounds great @wanasit.

So, just to confirm, the path forward is:

- Rename `forwardDateOnly` to `forwardWeekdays`
- Alias `forwardDayOnly` as `forwardDate`, provide a deprecation warning when used, remove when Chrono 2.0 is released to avoid breaking changes (semantic versioning)
- Add a result forwarding refiner activated by `forwardResults`

**Comment by wanasit** [comment]
Sorry. I didn't see your comment earlier. I ended up fixing it in c191a59.

I added ForwardDateRefiner. When we have `forwardDatesOnly` and `forwardDate` option, the refiner will try pushing the result into future by updating `year` or `week` part. It made code in parsers simpler too.

The option name is not announced the option officially in README.md. So we could rename it to something better. We could add `forwardWeekdays` and `forwardYear` as well, but I think it maybe too complicate.
